nexus/server.go

```diff
@@ -26,6 +26,21 @@ type handlerCtx struct {
 	links []Link
 }
 
+// WithHandlerContext returns a new context from a given context setting it up for being used for handler methods.
+// Meant to be used by frameworks, not directly by applications.
+func WithHandlerContext(ctx context.Context) context.Context {
+	return context.WithValue(ctx, handlerCtxKey, &handlerCtx{})
+}
+
+// HandlerLinks retrieves the attached links on the given handler context. The returned slice should not be mutated.
+// The context provided must be the context passed to the handler.
+func HandlerLinks(ctx context.Context) []Link {
+	hctx := ctx.Value(handlerCtxKey).(*handlerCtx)
```

Comment:
```suggestion
	hctx, ok := ctx.Value(handlerCtxKey).(*handlerCtx)
	if !ok {
		return nil
	}
```

Or if panic is intentional, document that it can panic if called outside of context (and possibly provide a way to know whether you're in a handler context). Same comment applies to `AddHandlerLinks`.

Reply:
I can document, the panic is intentional.

Reply:
We should consider a utility to let someone know a context is a handler context then. Basically general utility users should be able to know whether something they are about to call will panic (and therefore can prevent it with a conditional)

Reply:
Sure, that makes sense.